pkg/component/etcd/statefulset/values_helper.go

```diff
@@ -0,0 +1,683 @@
+		} else {
+			command = append(command, fmt.Sprintf("http://%s-local:%d/health", val.EtcdName, pointer.Int32Deref(val.ClientPort, defaultClientPort)))
+		}
+	}
+	return command
+}
+
+func getLivenessProbeCommand(val Values) []string {
+	command := []string{"" + "/bin/sh"}
+	command = append(command, "-ec")
+	command = append(command, "ETCDCTL_API=3")
+	command = append(command, "etcdctl")
+
+	if val.ClientUrlTLS != nil {
+
+		command = append(command, "--cert=/var/etcd/ssl/client/client/tls.crt")
+		command = append(command, "--key=/var/etcd/ssl/client/client/tls.key")
+		if dataKey := val.ClientUrlTLS.TLSCASecretRef.DataKey; dataKey != nil {
+			command = append(command, "--cacert=/var/etcd/ssl/client/ca/"+*dataKey)
+		}
+		command = append(command, fmt.Sprintf("--endpoints=https://%s-local:%d", val.EtcdName, pointer.Int32Deref(val.ClientPort, defaultClientPort)))
+	} else {
+		command = append(command, fmt.Sprintf("--endpoints=http://%s-local:%d", val.EtcdName, pointer.Int32Deref(val.ClientPort, defaultClientPort)))
+	}
+	command = append(command, "get")
+	command = append(command, "foo")
+	command = append(command, "--consistency=s")
+	return command
+}
+
+func getEtcdBackupCommand(val Values) []string {
+	command := []string{"" + "etcdbrctl"}
+	command = append(command, "server")
+
+	if val.BackupStore != nil {
+		command = append(command, "--enable-snapshot-lease-renewal=true")
+		command = append(command, "--delta-snapshot-lease-name="+val.DeltaSnapLeaseName)
+		command = append(command, "--full-snapshot-lease-name="+val.FullSnapLeaseName)
+	}
+
+	if val.DefragmentationSchedule != nil {
+		command = append(command, "--defragmentation-schedule="+*val.DefragmentationSchedule)
+	}
+
+	if val.FullSnapshotSchedule != nil {
+		command = append(command, "--schedule="+*val.FullSnapshotSchedule)
+	}
+
+	if val.GarbageCollectionPolicy != nil {
+
+		gbc := string(*val.GarbageCollectionPolicy)
+		command = append(command, "--garbage-collection-policy="+gbc)
+
+		if gbc == "LimitBased" {
+			command = append(command, "--max-backups=7")
+		}
+	} else {
+		command = append(command, "--garbage-collection-policy="+defaultGbcPolicy)
+		command = append(command, "--max-backups=7")
+	}
+
+	command = append(command, "--data-dir=/var/etcd/data/new.etcd")
+
+	if val.BackupStore != nil {
+		store, _ := utils.StorageProviderFromInfraProvider(val.BackupStore.Provider)
+		command = append(command, "--storage-provider="+store)
+		command = append(command, "--store-prefix="+string(val.BackupStore.Prefix))
+	}
+
+	var quota int64 = 8 * 1024 * 1024 * 1024 // 8Gi
+	if val.Quota != nil {
+		quota = val.Quota.Value()
+	}
+
+	command = append(command, "--embedded-etcd-quota-bytes="+fmt.Sprint(quota))
+
+	if pointer.BoolDeref(val.EnableProfiling, false) {
+		command = append(command, "--enable-profiling=true")
+	}
+
+	if val.BackupTLS != nil {
```

Comment:
The check here should be for `etcd.clientUrlTls`, since the endpoint being called is an etcd server endpoint.